psutil/arch/aix/net_connections.c

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2017, Arnon Yaari
+ * All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+/* Baded on code from lsof:
+ * http://www.ibm.com/developerworks/aix/library/au-lsof.html
+ * - dialects/aix/dproc.c:gather_proc_info
+ * - lib/prfp.c:process_file
+ * - dialects/aix/dsock.c:process_socket
+ * - dialects/aix/dproc.c:get_kernel_access
+*/
+
+#include "net_connections.h"
+#include <fcntl.h>
+#include <sys/types.h>
+#define _KERNEL 1
+#include <sys/file.h>
+#undef _KERNEL
+#include <stdlib.h>
+#include <sys/core.h>
+#include <sys/domain.h>
+#include <sys/un.h>
+#include <netinet/in_pcb.h>
+#include <arpa/inet.h>
+#include "net_kernel_structs.h"
+
+
+
+#define PROCINFO_INCR   (256)
+#define PROCSIZE        (sizeof(struct procentry64))
+#define FDSINFOSIZE     (sizeof(struct fdsinfo64))
+#define KMEM            "/dev/kmem"
+#define NO_SOCKET       (PyObject *)(-1)
+
+typedef u_longlong_t    KA_T;
+static int PSUTIL_CONN_NONE = 128;
+
+/* psutil_kread() - read from kernel memory */
+static int
+psutil_kread(
+    int Kd,             /* kernel memory file descriptor */
+    KA_T addr,          /* kernel memory address */
+    char *buf,          /* buffer to receive data */
+    size_t len) {       /* length to read */
+    int br;
+
+    if (lseek64(Kd, (off64_t)addr, L_SET) == (off64_t)-1)
+        return(1);
+    br = read(Kd, buf, len);
+    return((br == len) ? 0 : 1);
```

Comment:
If I understand this correctly you return `1` (meaning error) in case of len/size mismatch. Whereas the previous `lseek` and `read` calls set errno, this line (53) won't, so I think you need to do this:

```
if (br != len) {
    PyErr_SetString(PyExc_RuntimeError, 
                    "size mismatch when reading kernel memory fd");
    return 1;
}
return 0;
```

Also, since you set the exception in here, you should do the same for the 2 calls above (`lseek` and `read`) and do `PyErr_SetFromErrno(PyExc_OSError);`.
Of course the callers of this function should just return NULL.